test/ref_ops_test.cpp

```diff
@@ -6458,6 +6458,168 @@ TEST_CASE(quantizelinear)
+{
+    migraphx::program p;
+    auto* mm = p.get_main_module();
+    uint64_t seed(17);
+    size_t sample_size(200);
+
+    //      Shape of the random data
+    migraphx::shape rs{migraphx::shape::float_type, {{1, 2}, {2, sample_size + 1}}};
+    auto input = mm->add_parameter("Input_1", rs);
+
+    // Runtime randomization seed
+    migraphx::shape seed_shape{migraphx::shape::uint64_type, {1}};
+    auto seed_input = mm->add_parameter("Seed", seed_shape);
+
+    mm->add_instruction(migraphx::make_op("random_uniform", {}), seed_input, input);
+    p.compile(migraphx::make_target("ref"));
+
+    // Create a dummy input to hold the random data
+    migraphx::shape input_fixed_shape1{migraphx::shape::float_type, {sample_size}};
+
+    migraphx::parameter_map params0;
+    params0["Input_1"] = migraphx::argument(input_fixed_shape1);
+
+    std::vector<uint64_t> seed_data = {seed};
+    params0["Seed"]                 = migraphx::argument(seed_shape, seed_data.data());
+    auto result                     = p.eval(params0).back();
+
+    std::vector<float> result_vec(sample_size);
+    result.visit([&](auto output) { result_vec.assign(output.begin(), output.end()); });
+
+    // Compare result with the STL's mt19937 generator
+    std::mt19937 gen(seed);
+    std::uniform_real_distribution<> dis(0.0, 1.0);
+    std::vector<float> rand_samples(sample_size);
+    std::generate(rand_samples.begin(), rand_samples.end(), [&]() { return dis(gen); });
+    EXPECT(migraphx::verify::verify_range(result_vec, rand_samples, 100000));
+}
+
+TEST_CASE(random_uniform_and_seed_test)
+{
+    migraphx::program p;
+    auto* mm = p.get_main_module();
+
+    float control_seed = 0.0f;
+    size_t sample_size(20000);
+
+    //      Shape of the random data
+    migraphx::shape rs{migraphx::shape::float_type, {{1, 2}, {2, sample_size + 1}}};
+    auto input = mm->add_parameter("Input_1", rs);
+
+    // Runtime randomization seed
+    auto seed_input = mm->add_instruction(migraphx::make_op("random_seed"));
+    mm->add_instruction(migraphx::make_op("random_uniform"), seed_input, input);
+    p.compile(migraphx::make_target("ref"));
+
+    // Create a dummy input to hold the random data
+    migraphx::shape input_fixed_shape1{migraphx::shape::float_type, {sample_size}};
+
+    migraphx::parameter_map params0;
+    params0["Input_1"] = migraphx::argument(input_fixed_shape1);
+    auto result        = p.eval(params0).back();
+
+    std::vector<float> result_vec(sample_size);
+    result.visit([&](auto output) { result_vec.assign(output.begin(), output.end()); });
+
+    // Compare result with the STL's mt19937 generator.  Expected verify_range error ~0.4
+    std::mt19937 gen(control_seed);
+    std::uniform_real_distribution<> dis(0.0, 1.0);
+    std::vector<float> rand_samples(sample_size);
+    std::generate(rand_samples.begin(), rand_samples.end(), [&]() { return dis(gen); });
+    EXPECT(migraphx::verify::verify_range(result_vec, rand_samples, 1.e8));
+}
```

Comment:
Your second sentence is correct.  This is a rather special testing situation since `random_seed` is, to my knowledge, the only operation whose only use is as a prerequisite for a specific other operation.  After giving it some more thought, I don't think comparing the result against a second random distribution is such a bad idea.  It should give us an error if the code breaks.

Reply:
After more discussion, I removed the comparison against a second random distribution.  The test is just that it runs both ops without failing.